api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -532,7 +533,14 @@ public void setVisit(Visit visit) {
 		
 		return encounterProviders.stream()
 				.filter(ep -> includeVoided || !ep.getVoided())
-				.collect(Collectors.groupingBy(EncounterProvider::getEncounterRole, Collectors.mapping(EncounterProvider::getProvider, Collectors.toSet())));
+				.collect(Collectors.toMap((ep) -> (ep).getEncounterRole(),
```

Comment:
@pushpa446 and I talked about this. We thought this was a bug because it appeared when we updated bahmni-core to build against the latest openmrs-core, but we now realize that actually our Bahmni test data was invalid. We will cancel this PR and ticket, and just fix our Bahmni test data to include an encounter role.